src/main/java/seedu/address/model/person/Task.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    /**
+     * Constructor of Task given date, name and time represented by hour and min.
+     * @param date Date of task.
+     * @param startHour StartHour of task.
+     * @param startMin StartMin of task.
+     * @param endHour EndHour of task.
+     * @param endMin EndMin of task.
+     * @param taskName Name of task.
+     * @throws InvalidTimeException Throws if the given hour and min are invalid.
+     */
+    public Task(Date date, int startHour, int startMin, int endHour, int endMin, String taskName)
+            throws InvalidTimeException {
+        this.date = date;
+        this.startTime = new Time(startHour, startMin);
+        this.endTime = new Time(endHour, endMin);
+        this.taskName = taskName;
+    }
+
+    /**
+     * Constructor of Task given date, and time represented by hour and min.
+     * @param date Date of task.
+     * @param startHour StartHour of task.
+     * @param startMin StartMin of task.
+     * @param endHour EndHour of task.
+     * @param endMin EndMin of task.
+     * @throws InvalidTimeException Throws if the given hour and min are invalid.
+     */
+    public Task(Date date, int startHour, int startMin, int endHour, int endMin) throws InvalidTimeException {
+        this.date = date;
+        this.startTime = new Time(startHour, startMin);
+        this.endTime = new Time(endHour, endMin);
+    }
+
+    /**
+     * Sets name of the task.
+     * @param taskName Name of the task.
+     */
+    public void setTaskName(String taskName) {
+        this.taskName = taskName;
+    }
+
+    /**
+     * Returns the startTime of the task.
+     * @return The startTime of the task.
+     */
+    public Time getStartTime() {
+        return startTime;
+    }
+
+    /**
+     * Returns the endTime of the task.
+     * @return The endTime of the task.
+     */
+    public Time getEndTime() {
+        return endTime;
+    }
+
+    /**
+     * Gets the duration of this task in String.
+     *
+     * @return the duration of this task in String.
+     */
+    public String getDurationHour() {
+        return endTime.comparedToOther(startTime);
+    }
+
+    /**
+     * Finds whether a time is inside the period of this task.
+     *
+     * @param time The time.
+     * @return Whether the time is inside the period of this task.
+     */
+    public boolean inInterval(Time time) {
+        return time.compareTo(startTime) >= 0 && time.compareTo(endTime) <= 0;
+    }
+
+    @Override
+    public String toString() {
+        return taskName + ": " + "From " + getStartTime() + " to " + getEndTime();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+        if (!(other instanceof Task)) {
+            return false;
+        }
+        Task otherTask = (Task) other;
+        return toString().equals(otherTask.toString());
+    }
+
+    @Override
+    public int hashCode() {
+        return toString().hashCode();
+    }
+}
```

Comment:
In what scenario will this function be used?

Reply:
I just think that in every class it's better to override its hashCode() function. Maybe it will be useful later.